pymc_marketing/mmm/components/adstock.py

```diff
@@ -0,0 +1,174 @@
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+"""Adstock transformations for MMM.
+
+Each of these transformations is a subclass of
+:class:`pymc_marketing.mmm.components.adstock.AdstockTransformation`
+and defines a function that takes a time series and returns the adstocked
+version of it. The parameters of the function are the parameters
+of the adstock transformation.
+
+Examples
+--------
+Create a new adstock transformation:
+
+.. code-block:: python
+
+    class MyAdstock(AdstockTransformation):
+        def function(self, x, alpha):
+            return x * alpha
+
+        default_priors = {"alpha": {"dist": "HalfNormal", "kwargs": {"sigma": 1}}}
+
+"""
+
+import warnings
+
+from pymc_marketing.mmm.components.base import Transformation
+from pymc_marketing.mmm.transformers import (
+    ConvMode,
+    WeibullType,
+    delayed_adstock,
+    geometric_adstock,
+    weibull_adstock,
+)
+
+
+class AdstockTransformation(Transformation):
+    """Subclass for all adstock functions.
+
+    In order to use a custom saturation function, inherit from this class and define:
+
+    - `function`: a function that takes x to adstock x
+    - `default_priors`: dictionary with priors for every parameter in function
+
+    Consider the predefined subclasses as examples.
+
+    """
+
+    prefix: str = "adstock"
+
+    def __init__(
+        self,
+        l_max: int,
+        normalize: bool = True,
+        mode: ConvMode = ConvMode.After,
+        priors: dict | None = None,
+        prefix: str | None = None,
+    ) -> None:
+        self.l_max = l_max
+        self.normalize = normalize
+        self.mode = mode
+
+        super().__init__(priors=priors, prefix=prefix)
+
+
+class GeometricAdstock(AdstockTransformation):
+    """Wrapper around geometric adstock function.
+
+    For more information, see :func:`pymc_marketing.mmm.transformers.geometric_adstock`.
+
+    """
+
+    def function(self, x, alpha):
+        return geometric_adstock(
+            x, alpha=alpha, l_max=self.l_max, normalize=self.normalize, mode=self.mode
+        )
+
+    default_priors = {"alpha": {"dist": "Beta", "kwargs": {"alpha": 1, "beta": 3}}}
+
+
+class DelayedAdstock(AdstockTransformation):
+    """Wrapper around delayed adstock function.
+
+    For more information, see :func:`pymc_marketing.mmm.transformers.delayed_adstock`.
+
+    """
+
+    def function(self, x, alpha, theta):
+        return delayed_adstock(
+            x,
+            alpha=alpha,
+            theta=theta,
+            l_max=self.l_max,
+            normalize=self.normalize,
+            mode=self.mode,
+        )
+
+    default_priors = {
+        "alpha": {"dist": "HalfNormal", "kwargs": {"sigma": 1}},
```

Comment:
The `alpha` of the adstock functions has to be between zero and one right? We should use a Beta distribution as default.

Reply:
```suggestion
        "alpha": {"dist": "Beta", "kwargs": {"alpha": 1, "beta": 3}},
```